AU2/plugins/custom_plugins/ScoringPlugin.py

```diff
@@ -33,6 +42,66 @@
           errors="ignore") as F:
     OPENSEASON_PAGE_TEMPLATE = F.read()
 
+# TODO: split up go generate table and row templates based on options
+# LHS is table header, RHS is template string for table cell values
+# note the order here determines the order in which columns are displayed
+STATS_COLUMN_TEMPLATES = {
+    "Position": "{RANK}",
+    "Died at": "{DEATHS}",
+    "Real Name": "{NAME}",
+    "Pseudonym": "{PSEUDONYMS}",
+    "Number of Attempts": "{ATTEMPTS}",
+    "Number of Kills": "{KILLS}",
+    "Conkers Score": "{CONKERS}",
+    "Score": "{SCORE}"
+}
+
+
+def stats_row_template(columns: Iterable[str]) -> str:
+    """Generate a row template for the stats page from the given columns"""
+    return "<tr>" + "".join(f"<td>{STATS_COLUMN_TEMPLATES[col]}</td>" for col in columns) + "</tr>"
+
+
+def stats_table_template(columns: Iterable[str]) -> str:
+    """Generate a table template for the stats page from the given columns"""
+    return """
+<table xmlns="" class="playerlist">
+<tr>""" + "".join(f"<th>{header}</th>" for header in columns) + """</tr>
+  {ROWS}
+</table
+"""

```

Comment:
A few things here:
1) Your final table tag is unclosed.
2) HTML doesn't care about the newlines here
3) The indentation is a little hard to follow

How about something like this?

```suggestion
    return (
        '<table xmlns="" class="playerlist"><tr>'
        + ''.join(f'<th>{header}</th>' for header in columns)
        + '</tr> {ROWS} </table>'
   )
```

Reply:
Looks good! I don't know how I missed the unclosed `table` tag...